### ISSUE TITLE
Firmware update on ITC

### ISSUE THREAD

**Issue by Peterkovacs95** [torriem/waas_steer#5]
Hi!

Do you have the zip file for the update of the reciver?
Unfortunately the link in the AgTalk post is not accesibile anymore.

Cheers, Peter

**Comment by torriem** [comment]
The link still works. https://isgsupport.deere.com/dtac-files/3.73J.zip

**Comment by Peterkovacs95** [comment]
Got it, thank you for the fast response.!

**Comment by Peterkovacs95** [comment]
Hey!

I managed to update my reciver, getting the EGNOS signal no problem, but when i plug in the Teensy 4.0 board the reciver is not recognised by the screen. My setup looks like this:
![image](https://github.com/torriem/waas_steer/assets/127775074/5bf7d088-1561-4555-92ca-f70a329569b9)

Any advice or idea?

**Comment by Peterkovacs95** [comment]
![IMG_3762](https://github.com/torriem/waas_steer/assets/127775074/626e9644-9ead-46dd-9883-a841991068d3)
![IMG_3758](https://github.com/torriem/waas_steer/assets/127775074/64812b4b-ce1b-4452-856c-ef74a8ca78f6)
![IMG_3757](https://github.com/torriem/waas_steer/assets/127775074/9e0a6404-02ed-46cb-aa8d-20892c9d3dc8)

**Comment by torriem** [comment]
What does it look like on the main Autotrac guidance screen?  An autotrac be enabled?  Going into the Starfire screen will always just show EGNOS because that's all the receiver is getting.  The SF1 faking is only visible to autotrac and on the guidance screen.

**Comment by Peterkovacs95** [comment]
The problem is that, when de board is plugged in the monitor do not recognise the itc. As soo as i take the board out of the system the reciver gathers the egnos signal without problem.
![image](https://github.com/torriem/waas_steer/assets/127775074/c7752dd3-a6e2-4bc5-b22b-938fb2ead827)
![image](https://github.com/torriem/waas_steer/assets/127775074/207ec962-9b5c-4bc4-ad3c-b78de8616d78)

**Comment by Peterkovacs95** [comment]
There are two can terminators in the system one is ont the ATU300 hatness and one is on the green star harness near the reciver plug. If i pull out the one near the reciver the monitor can not recognize the globe even without the teensy. If i unplug the terminator from the atu harness the reciver can be acessed only if the teensy is not plugged in.

**Comment by torriem** [comment]
Oh I see. Yes it probably is a termination issue,  but I don't know where or how to fix it.  In the Deere monitor there is a place where you can look at some CAN bus parameters including voltage.  Would be interesting to compare the voltage without the teensy and with the teensy.  

Have you tried different terminator jumper settings on the teensy?  I see both sides are terminated on your board. Maybe try removing the termination on one side at a time and see if that changes things?

**Comment by Peterkovacs95** [comment]
So, what i did so far:

![IMG_0333](https://github.com/torriem/waas_steer/assets/127775074/6c3578e2-a7c1-45ec-8eae-5f9e2a7f8acf)
![IMG_0334](https://github.com/torriem/waas_steer/assets/127775074/6b062efe-b852-4cb5-8b5d-4dd3a5ae9d97)


I managed to fidure out these CAN terminators are not static 120 ohm resistors, it is using current to modify the resistance somehow.
I pulled out the CAN terminator form the reciver side then Bridged the CAN H and L wires with a piece of copper wire. I resoldered the J2 jumper and the 120 ohm resistance appeard at one side of the 12 pin connector. Then I resoldered the J1 jumper and only 1 ohm resistance apeard on the other side. I also tried to remove both J1 and J2 jumpers but no luck. Tomorrow i will try to remove the other JD CAN terminator from the atu harness and bridge those CAN wires as well. I hope it will work otherwise I do not know any other possible solution. I will check if i can find the CAN voltages in the monitor as well.

**Comment by torriem** [comment]
Right.  Deere considers them to be passive terminators, and they do fail sometimes.  But I don't know how they work, unfortunately.

**Comment by Peterkovacs95** [comment]
Well, the problem was mutch simpler than i tought. In the 12 pin male connector the Nr. 12 pin was slid out from the housing of the connector. Both terminators are plugged back to the harness and both terminators are soldered back to the board. Job Done. Thanks for your assistance and the code as well.

**Comment by torriem** [comment]
Awesome!  You're very welcome.  Glad you were able to get it working.
